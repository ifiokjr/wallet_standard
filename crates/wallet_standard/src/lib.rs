#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/readme.md"))]

pub use error::*;
pub use experimental::*;
#[cfg(feature = "solana")]
pub use solana::*;
pub use standard::*;
pub use types::*;

mod error;
mod experimental;
#[cfg(feature = "solana")]
mod solana;
mod standard;
mod types;

pub mod prelude {
	pub use super::ExperimentalDecryptOutput;
	pub use super::ExperimentalEncryptOutput;
	pub use super::IntoWalletError;
	pub use super::StandardConnectOutput;
	pub use super::Wallet;
	pub use super::WalletAccountInfo;
	pub use super::WalletError;
	pub use super::WalletExperimentalDecrypt;
	pub use super::WalletExperimentalEncrypt;
	pub use super::WalletInfo;
	pub use super::WalletResult;
	pub use super::WalletStandard;
	pub use super::WalletStandardConnect;
	pub use super::WalletStandardDisconnect;
	#[cfg(feature = "solana")]
	pub use super::solana::prelude::*;
}

#[cfg(feature = "solana")]
#[cfg(test)]
mod tests {
	use solana_sdk::message::Message;
	use solana_sdk::signature::Signature;
	use solana_sdk::transaction::Transaction;
	use solana_sdk::transaction::VersionedTransaction;

	use super::*;

	#[test]
	fn versioned_transaction_implements_solana_sign_transaction_output() {
		let transaction =
			VersionedTransaction::from(Transaction::new_unsigned(Message::new(&[], None)));
		insta::assert_json_snapshot!(transaction.signed_transaction_bytes());
	}

	#[test]
	fn transaction_implements_solana_sign_transaction_output() {
		let transaction = Transaction::new_unsigned(Message::new(&[], None));
		insta::assert_json_snapshot!(transaction.signed_transaction_bytes());
	}

	#[test]
	fn signature_implements_signature_output() {
		let signature = Signature::default();
		insta::assert_json_snapshot!(signature.try_signature());
	}
}
