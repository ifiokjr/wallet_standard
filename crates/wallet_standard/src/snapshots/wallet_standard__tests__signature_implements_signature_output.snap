---
source: crates/wallet_standard/src/lib.rs
expression: signature.try_signature()
snapshot_kind: text
---
{
  "Ok": [
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0
  ]
}
